File: services/inventory-service/src/main/java/io/minishop/inventory/exception/OutOfStockException.java
package io.minishop.inventory.exception;

public class OutOfStockException extends RuntimeException {
    private final Long productId;
    private final int requested;
    private final int available;

    public OutOfStockException(Long productId, int requested, int available) {
        super("Out of stock for product " + productId + " (requested=" + requested + ", available=" + available + ")");
        this.productId = productId;
        this.requested = requested;
        this.available = available;
    }

    public Long getProductId() { return productId; }
    public int getRequested() { return requested; }
    public int getAvailable() { return available; }
}

File: services/inventory-service/src/main/kotlin/io/minishop/inventory/exception/OutOfStockException.kt
package io.minishop.inventory.exception

class OutOfStockException(
    productId: Long,
    requested: Int,
    available: Int,
) : RuntimeException(
    "Out of stock for product $productId (requested=$requested, available=$available)",
) {

    @get:JvmName("getProductId")
    val productId: Long = productId

    @get:JvmName("getRequested")
    val requested: Int = requested

    @get:JvmName("getAvailable")
    val available: Int = available
}
